vector_helpers_test.go

```diff
@@ -0,0 +1,381 @@
+package cimgui
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestImVec2_Add(t *testing.T) {
+	type fields struct {
+		X float32
+		Y float32
+	}
+
+	type args struct {
+		another ImVec2
+	}
+
+	tests := []struct {
+		name   string
+		fields fields
```

Comment:
Just make `a Vec2` and `b Vec2`. `fields` and `args` is a bad name.
The test names are noisy - I'd say it's better to drop them and make them comments. This applies to all the tests.

Reply:
I'd leave them as is because that template is automatically generated by goland

Reply:
"that template is automatically generated by goland"
And? This doesn't mean it's good code.

Reply:
> "that template is automatically generated by goland"
> And? This doesn't mean it's good code.

ofc it doesn't, I just think that it is commonly used :smile:
but let's deiscuss it:

> The test names are noisy - I'd say it's better to drop them and make them comments. This applies to all the tests.

I can't agree, lets imagine that one of tests fails. If test didn't have a name, you wouldn't know which exactly failed. The name (it could be anything unique e.g. "red fox" or so...) helps you identify the faulty place, doesn't it?

> Just make a Vec2 and b Vec2. fields and args is a bad name.

You're probably right. It isn't the best of names, but it makes sense after considering it:
"fields" stands from struct (reciver's) fields
"args" are arguments of method.

If you insist, I can change it, however IMO it doesn't really matter.

Reply:
"ofc it doesn't, I just think that it is commonly used 😄"
1. I didn't see it much in real code. And if I did, I would get really annoyed
2. I think it's better to do what Go's standard library does with tests - do minimal amount of "magic" for tests. If the test is not trivial, it should be moved to a new unit test. If it's trivial, then it's easy to see which input caused which output and which case failed. "'add positive' test failed!" has much less meaning than "test failed: Vec{10,20}.Add(Vec2{40,30}) - expected Vec2{50,50}, got Vec2{10, 10}"

See [here](https://cs.opensource.google/go/go/+/master:src/strings/search_test.go;l=13) for example. 

Note that "expected X, got Y" string should include input arguments as well. 

`Add() - expected X got Y` is not as good as `Vec2{X,Y}.Add(Vec2{X2,Y2}) - expected X, got Y`

Reply:
ok, I've removed argss/fields and replaced with simple `p`/`q`.
However, after looking on the code I've noticed thet `name` has one more reason for keeping them there.
the script runs a sub-test wit `t.Run` and it needs test's name.